core/src/main/java/org/kiwix/kiwixmobile/core/main/ToolbarScrollingKiwixWebView.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getToolbarHeight
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import kotlin.math.max
+import kotlin.math.min
+
+@SuppressLint("ViewConstructor")
+class ToolbarScrollingKiwixWebView(
+  context: Context,
+  callback: WebViewCallback,
+  attrs: AttributeSet,
+  nonVideoView: ViewGroup,
+  videoView: ViewGroup,
+  webViewClient: CoreWebViewClient,
+  private val toolbarView: View,
+  private val bottomBarView: View,
+  sharedPreferenceUtil: SharedPreferenceUtil,
+  private val parentNavigationBar: View?
```

Comment:
perhaps I should have been firmer in my conviction. Did you try making this default to null and using `@JVMOverloads` to avoid writing the second constructor?

Reply:
yes, I made the secondary constructor to null.

Reply:
> Optionally we could also have null as the default argument and let @JvmOverloads dupe out the constructors for us

with `jvmOverloads` we can avoid having to write 2 constructors. It is basically how you hoodwink java into working well with default arguments

Reply:
Thank you @macgills . @JvmOverloads are super cool. Next time I'll remember that.